airflow/utils/timezone.py

```diff
@@ -273,3 +279,24 @@ def _format_part(key: str) -> str:
     if not joined:
         return "<1s"
     return joined
+
+
+@lru_cache(maxsize=None)
+def parse_timezone(name: str | int) -> Timezone | FixedTimezone:
```

Comment:
just `Timezone` should be sufficient? What about the parameter `name`? That looks a bit awkward if you are checking for an `int`. What about `tz` and making it part of the docstring?

Reply:
> just Timezone should be sufficient? 

https://github.com/apache/airflow/pull/34744#discussion_r1347070794

> What about the parameter name? That looks a bit awkward if you are checking for an int. What about tz and making it part of the docstring?

This is basically reimplementation of pendulum parse function, so I keep the same attribute `name` but I have no objection if we would like to rename it

Reply:
I'd be for leaving it as is - with name. The int is really kind of backcompat - I can't see anyone  using int to specify it in the future and while we want to keep positionally specified parameter as int for backcompat and possibly some interna usage, the keyword should be `name` - that would also guide the user who uses keyword to use actual name not I'd if timezone where using keyword parameter.

It's a strong indicator for the user of that function that they are supposed to use string.